qemu/tests/chardev_tls_encryption.py

```diff
@@ -0,0 +1,109 @@
+        port = str(utils_misc.find_free_ports(5000, 9999, 1, hostname)[0])
+
+        # Scenario 1
+        gnutls_cmd_server = params.get("gnutls_cmd_server")
+        if gnutls_cmd_server:
+            gnutls_cmd_server = gnutls_cmd_server % port
+            params["extra_params"] = params["extra_params"] % (hostname, port)
+            error_context.context("Run gnutls server ...", logging.info)
+            tls_server = aexpect.run_bg(gnutls_cmd_server)
+            params['start_vm'] = 'yes'
+            vm_name = params['main_vm']
+            error_context.context(
+                "Launch QEMU with a serial port as TLS client", logging.info)
+            env_process.preprocess_vm(test, params, env, vm_name)
+            if not utils_misc.wait_for(
+                    lambda: expected_msg in tls_server.get_output(),
+                    first=5, timeout=15):
+                test.fail("TLS server can't connect client succssfully.")
+
+        # Scenario 2
+        gnutls_cmd_client = params.get("gnutls_cmd_client")
+        if gnutls_cmd_client:
+            gnutls_cmd_client = gnutls_cmd_client % (port, hostname)
+            params["extra_params"] = params["extra_params"] % (hostname, port)
+            params['start_vm'] = 'yes'
+            vm_name = params['main_vm']
+            error_context.context(
+                "Launch QEMU with a serial port as TLS server", logging.info)
+            env_process.preprocess_vm(test, params, env, vm_name)
+            error_context.context(
+                "Run gnutls client to connect TLS server", logging.info)
+            tls_client = aexpect.run_bg(gnutls_cmd_client)
+            if not utils_misc.wait_for(
+                    lambda: expected_msg in tls_client.get_output(),
+                    first=5, timeout=15):
+                test.fail("TLS client can't connect server succssfully.")
+
+        # Scenario 3:
+        guest_cmd = params.get("guest_cmd")
+        if guest_cmd:
+            tmp_file = params["tmp_file"]
+            params["start_vm"] = "yes"
+            vms = params.get("vms").split()
+            params["extra_params"] = params["extra_params_%s"
+                                            % vms[0]] % (hostname, port)
+            error_context.context(
+                "Launch QEMU with a serial port as TLS server", logging.info)
+            env_process.preprocess_vm(test, params, env, vms[0])
+            vm1 = env.get_vm(vms[0])
+            session_vm1 = vm1.wait_for_login()
+            session_vm1.cmd(guest_cmd)
+            params["extra_params"] = params["extra_params_%s"
+                                            % vms[1]] % (hostname, port)
+            error_context.context(
+                "Launch QEMU with a serial port as TLS client", logging.info)
+            env_process.preprocess_vm(test, params, env, vms[1])
+            vm2 = env.get_vm(vms[1])
+            if not utils_misc.wait_for(
```

Comment:
Eh~   just noticed,  when use `cat /root/client-log` multiple times,  it read the full file every time but not the increment.  This is not same with the another 2 scenario above which readout the data every time.
Suggest:
```
            session_vm1.cmd('cat /dev/ttyS0')
            params["extra_params"] = params["extra_params_%s"
                                            % vms[1]] % (hostname, port)
            error_context.context(
                "Launch QEMU with a serial port as TLS client", logging.info)
            env_process.preprocess_vm(test, params, env, vms[1])
            try:
                session_vm1.read_until_output_matches(expected_msg, timeout=15)
            except aexpect.ExpectError:
                test.fail("Can't connect TLS client inside TLS server guest.")
            vm2 = env.get_vm(vms[1])
            session_vm1.close
            ...
```

Reply:
thanks and update.

Test result.
(1/1) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.1.x86_64.io-github-autotest-qemu.chardev_tls_encryption.guest_to_guest: PASS (70.46 s)